File: src/org/cappuccino_project/ide/intellij/plugin/psi/interfaces/ObjJHasBlockStatement.java
package org.cappuccino_project.ide.intellij.plugin.psi.interfaces;

import org.cappuccino_project.ide.intellij.plugin.psi.ObjJBlock;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.List;

public interface ObjJHasBlockStatement extends ObjJHasBlockStatements{
    @Nullable
    ObjJBlock getBlock();

    @NotNull
    default List<ObjJBlock> getBlockList() {
        return getBlock() != null ? Collections.singletonList(getBlock()) : Collections.emptyList();
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/psi/interfaces/ObjJHasBlockStatement.kt
package org.cappuccino_project.ide.intellij.plugin.psi.interfaces

import org.cappuccino_project.ide.intellij.plugin.psi.ObjJBlock

import java.util.Collections

interface ObjJHasBlockStatement : ObjJHasBlockStatements {
    val block: ObjJBlock?

    override val blockList: List<ObjJBlock>
        get() = if (block != null) listOf<ObjJBlock>(block) else emptyList()
}
